packages/local_auth/local_auth_ios/ios/Classes/FLTLocalAuthPlugin.m

```diff
@@ -216,41 +220,35 @@ - (void)handleAuthReplyWithSuccess:(BOOL)success
     result(@YES);
   } else {
     switch (error.code) {
+      case LAErrorSystemCancel:
+        if ([arguments[@"stickyAuth"] boolValue]) {
+            self->_lastCallArgs = arguments;
+            self->_lastResult = result;
+        } else {
+            result(@NO);
+        }
+        return;
       case LAErrorPasscodeNotSet:
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-      // TODO(stuartmorgan): Remove the pragma and s/TouchID/Biometry/ in these constants when
-      // iOS 10 support is dropped. The values are the same, only the names have changed.
-      case LAErrorTouchIDNotAvailable:
-      case LAErrorTouchIDNotEnrolled:
-      case LAErrorTouchIDLockout:
-#pragma clang diagnostic pop
+      case LAErrorAuthenticationFailed:
+      case LAErrorBiometryNotAvailable:
+      case LAErrorBiometryNotEnrolled:
+      case LAErrorBiometryLockout:
       case LAErrorUserFallback:
+      default:
```

Comment:
when will this default be called?

Reply:
So I added the default so it would catch any unknown error codes and throw an error - the original issue was happening because an error code was thrown that wasnt listed and it was failing silently.  Should I change my approach?

Reply:
> the original issue was happening because an error code was thrown that wasnt listed and it was failing silently. 

can you inspect the error code and explicitly list here? The `default` case shadows all potential future cases and may cause hard-to-find bugs.

Reply:
the error code was 
https://github.com/flutter/plugins/blob/abb8e681c414c6d396ae8ccc0bfc9a1293b1602c/packages/local_auth/local_auth_ios/ios/Classes/FLTLocalAuthPlugin.m#L232

So its already listed
Should I remove the default case then?

Reply:
I agree, we should handle all enums and remove the `default` to detect (via compilation warning) when this problem happens again the next time we update the SDK.